### PR TITLE
TPIU: swo_supports: make struct fields public, improve documentation

### PR DESCRIPTION
**Pull request by tmplt**
A few small changes that I missed the first time around when the TPIU API was expanded.

### CODE REVIEW

File: src/peripheral/tpiu.rs [hunk 1]
```diff
@@ -148,8 +149,8 @@ impl TPIU {
     /// Reads the supported trace output modes and the minimum size of
     /// the TPIU FIFO queue for trace data.
     #[inline]
-    pub fn swo_supports() -> SWOSupports {
-        let _type = unsafe { (*Self::ptr())._type.read() };
+    pub fn swo_supports(&self) -> SWOSupports {
```
Comment (TDHolmes) [inline src/peripheral/tpiu.rs:152]: Per [this comment](https://github.com/rust-embedded/cortex-m/issues/183#issuecomment-579430492), _should_ self be added? Seems not Needed here

Comment (tmplt) [reply src/peripheral/tpiu.rs:152]: Indeed not required. I glossed over that comment. I'll revert it.

**Review by TDHolmes — APPROVED** [review]
Not sure what my review is worth, but LGTM :p

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - DWT: add `configure` API for address, cycle count comparison (#342, #367).
 - ITM: add `configure` API (#342).
 - TPIU: add API for *Formatter and Flush Control* (FFCR) and *Selected Pin Control* (SPPR) registers (#342).
+- TPIU: add `swo_supports` for checking what SWO configuratios the target supports. (#381)
```
Comment (thejpster) [inline CHANGELOG.md:17]: Typo

**Review by thejpster — APPROVED** [review]

**Pull request merged**